[docs] Revise Joy UI "Radio" page (#35893)
mui/material-ui: docs/data/joy/components/radio/radio.md

> # Radio Group

Reviewer: Let's call this page "Radio" since it covers both buttons and groups.

```suggestion
# Radio
```

<p class="description">The Radio Group enables the user to select one option from a set.</p>

## Introduction

Radio buttons let users make a mututally exclusive choice (e.g., this or that). Only one selection is allowed from the available set of options.

To provide the user with multiple selection options, use Checkbox instead.

> If available options can be collapsed, consider using a Select component to conserve space.

Author: Adding that last line in the callout

{{"demo": "RadioButtons.js"}}

## Customization

### Label

Use `label` prop to label the Radio button.

{{"demo": "RadioButtonLabel.js"}}

For more complex layouts, compose a Radio button with Form Control, Form Label, and Form Helper Text (optional).

{{"demo": "RadioButtonControl.js"}}

### Position

To swap the label and Radio position, use the CSS property `flex-direction: row-reverse`.

{{"demo": "RadioPositionEnd.js"}}

### Radio group

The Radio Group component is the ideal wrapper for multiple Radio components as it provides a tailored API for Radio button grouping and proper keyboard navigation accessibility support.

{{"demo": "RadioButtonsGroup.js"}}

### Controlled

To control what the Radio button is selecting, use the `value` and `onChange` props.

{{"demo": "ControlledRadioButtonsGroup.js"}}

### Focus outline

The focus outline, by default, wraps both the Radio and its label.
To change that, target the `radioClasses.radio` class and add `position: 'relative'`.

{{"demo": "RadioFocus.js"}}

### Overlay

To make the Radio button's container clickable, use the `overlay` prop.

> Applying `overlay` to the parent Radio Group component will pass the prop to each child Radio button inside the Radio Group.

Reviewer: Happy medium between the first and second draft. :)

```suggestion
You can also apply this prop directly to a Radio Group when present, which will pass the prop to each individual Radio button nested within.
```

> ### Icon

Reviewer: ```suggestion
### Custom icons
```

> Radio, by default, comes without an unchecked component.
> To add an icon to both unchecked and checked states, use the `uncheckedIcon` and `checkedIcon` props.

Reviewer: This section doesn't do a great job of explaining this feature IMO. Here's how I'd rewrite it.

```suggestion
Use the `checkedIcon` and/or `uncheckedIcon` props to add custom icons to the Radio button that correspond to each state, respectively.
The demo below shows how to apply a custom `checkedIcon` to the Radio button that sits on the corner of each Sheet:
```

{{"demo": "IconsRadio.js"}}

> ### Without an icon

Reviewer: ```suggestion
### No icon
```

> To communicate the checked and unchecked states with different properties, such as border or background color, use the `disableIcon` prop to remove the default icon.

Reviewer: This one could also use a slightly different explanation.

```suggestion
Use the `disableIcon` prop to remove the Radio button's icon.
In this case, you'll need to use CSS properties like border and background color to communicate the state of the Radio button, as shown in the demo below:
```

### Alignment buttons

> Provide an icon as a label to the Radio to make the buttons concise. You need to provide `aria-label` to the input slot for users who rely on screen readers.

Reviewer: ```suggestion
This example uses icons as labels for a group of Radio buttons to recreate the form and function of [Toggle Buttons](https://mui.com/material-ui/react-toggle-button/).
In this case, you must provide an `aria-label` to the input slot for users who rely on screen readers.
```

{{"demo": "ExampleAlignmentButtons.js"}}

### Payment methods

> Mix Radio buttons with the [List](/joy-ui/react-list/)-related components to create a commonly seen vertical or horizontal payment method list.

Reviewer: ```suggestion
Mix Radio buttons with the [List](/joy-ui/react-list/) components to create a vertical or horizontal payment method list.
```

{{"demo": "ExamplePaymentChannels.js"}}

### E-commerce product attributes

> This example demonstrates complex customization using the Sheet component as a container for the Radios.

Reviewer: ```suggestion
This example demonstrates complex customization using the Sheet component as a container for the Radio buttons.
```

The focus outline is customized to be smaller, and the color changes based on the value.

> The check icon's color inherits the Radio button's `solid` variant, so that we don't need to handpick a color that has enough contrast with the background.

Reviewer: ```suggestion
The check icon's color inherits the Radio button's `solid` variant, so you don't need to handpick a color that contrasts enough with the background.
```

{{"demo": "ExampleProductAttributes.js"}}

## Accessibility

Here are a few tips to make sure you have an accessible Radio button component:

- Every Form Control should have proper labels.
  This includes Radio buttons, Checkboxes, and Switches.
  In most cases, this is done by using the Form Control and Form Label element.
- When a label can't be used, make sure to add an attribute, such as `aria-label`, `aria-labelledby`, and/or `title`, directly on the input component.
  You can also use the `inputProps` prop to add them.

```jsx
<Radio
  value="radioA"
  slotProps={{
    input: {
      'aria-label': 'Radio A',
    },
  }}
/>
```

Visit the [WAI-ARIA documentation](https://www.w3.org/WAI/ARIA/apg/patterns/radiobutton/) for more details.

## Anatomy

The Radio Group component is composed of a root `<div>` element that can wrap multiple Radio components.

```html
<div class="JoyRadioGroup-root">

>   <!-- radio components here -->

Reviewer: ```suggestion
  <!-- Radio components here -->
```

</div>
```

> The Radio component is composed of a root `<span>`, with further nested `<span>` for the radio, icon, action (with a nested `<input>`), and its associated `<label>`.

Reviewer: ```suggestion
The Radio component is composed of a root `<span>`, with further nested `<span>` elements for the radio, icon, action (with a nested `<input>`), and its associated `<label>`.
```
